src/monmove.cpp

```diff
@@ -1445,3 +1445,21 @@ int monster::turns_to_reach( int x, int y )
 
     return int( turns + .9 ); // Halve (to get turns) and round up
 }
+
+void monster::on_idle()
+{
+    if( has_flag( MF_EGGLAYING ) && !has_effect( effect_laid_egg ) ) {
+        const int range = 15;
+        // Loop to find the closest furniture with NEST flag and set it as destination
+        for( tripoint &p : closest_tripoints_first( range, pos() ) ) {
+            if( sees( p ) && g->m.has_flag( TFLAG_NEST, p ) ) {
+                set_dest( p );
```

Comment:
You should probably leave the loop after this via `break`. Currently the game would go on and check all the other tiles within range as well. The monster would essentially go to the furthest visible point.